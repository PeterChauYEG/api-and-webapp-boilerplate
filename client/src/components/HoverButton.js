import React from 'react'

// import css
import '../CSS/HoverButton.css'
import 'font-awesome/css/font-awesome.css'

const HoverButton = (props) => {
  const {
    action,
    backgroundColor,
    icon,
  } = props

  return (
    <div className={`HoverButton ${backgroundColor}`}
      onClick={action} >
      <i className={`fa ${icon} fa-2x`}></i>
    </div>
  );
}

HoverButton.propTypes = {
  action: React.PropTypes.func.isRequired,
  backgroundColor: React.PropTypes.string.isRequired,
  icon: React.PropTypes.string.isRequired,
}

export default HoverButton;
