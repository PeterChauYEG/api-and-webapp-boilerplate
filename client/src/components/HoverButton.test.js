import React from 'react';

// import test dependencies 
import { shallow } from 'enzyme';

// import component
import HoverButton from './HoverButton'

it('renders without crashing', () => {
  const action = () => {}
  const backgroundColor = ""
  const icon = ""
  
  shallow(
    <HoverButton 
      action={action} 
      backgroundColor={backgroundColor} 
      icon={icon} 
      />
  );
});