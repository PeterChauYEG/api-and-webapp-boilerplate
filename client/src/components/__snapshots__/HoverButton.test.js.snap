exports[`test renders without crashing 1`] = `
<div
  className="HoverButton "
  onClick={[Function]}>
  <i
    className="fa  fa-2x" />
</div>
`;
